docs/deprecation.md

```diff
@@ -0,0 +1,28 @@
+# Deprecation Policy & Breaking Changes
+
+This documentation gives a short outline of our deprecation strategy in the
+project.
```

Comment:
I think it's useful to write something about the general policy here.

Reply:
I've written something but am not really happy with it.